tests/crypto_policy_test.go

```diff
@@ -0,0 +1,238 @@
+	portInt, err := strconv.ParseUint(port, 10, 16)
+	Expect(err).ToNot(HaveOccurred())
+	conn, err := portForwarder.Connect(&pod, uint16(portInt))
+	Expect(err).ToNot(HaveOccurred())
+	defer conn.Close()
+	tlsVersion, err := tlsConfig.MaxTLSVersionId()
+	if err != nil {
+		return err
+	}
+
+	certPool := x509.NewCertPool()
+
+	pemCert := getPemCertificate()
+	certPool.AppendCertsFromPEM(pemCert)
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			DialContext: func(ctx context.Context, network, addr string) (net.Conn, error) {
+				return conn, nil
+			},
+			TLSClientConfig: &tls.Config{CipherSuites: tlsConfig.CipherIDs(nil), MaxVersion: tlsVersion, RootCAs: certPool},
+		},
+	}
+
+	if subpath != "" {
+		subpath = "/" + subpath
+	}
+	_, err = client.Get(fmt.Sprintf("https://ssp-webhook-service.kubevirt.svc:%s%s", port, subpath))
+	return err
+}
+
+func testTLSEndpointAccessible(pod core.Pod, port, subpath string, tlsConfig clientTLSOptions) {
+	err := tryToAccessEndpoint(pod, port, subpath, tlsConfig)
+	Expect(err).ToNot(HaveOccurred(), "Can't access pod %s, at port %s, with tlsConfig %#v", pod.Name, port, tlsConfig)
+}
+
+func testTLSEndpointNotAccessible(pod core.Pod, port, subpath string, tlsConfig clientTLSOptions) {
+	err := tryToAccessEndpoint(pod, port, subpath, tlsConfig)
+	version, _ := tlsConfig.MaxTLSVersionId()
+	Expect(err).To(HaveOccurred(), "Should not have been able to access pod %s, at port %s, with tlsConfig %#v, %#v, %#v", pod.Name, port, tlsConfig, version, tlsConfig.CipherIDs(nil))
+}
+
+func testEndpointAccessabilityWithTLS(pod core.Pod, port, subpath string, allowedConfigs, disallowedConfigs []clientTLSOptions) {
+	for _, config := range allowedConfigs {
+		testTLSEndpointAccessible(pod, port, subpath, config)
+	}
+	for _, config := range disallowedConfigs {
+		testTLSEndpointNotAccessible(pod, port, subpath, config)
+	}
+}
+
+func testMetricsEndpoint(pod core.Pod, allowedConfigs, disallowedConfigs []clientTLSOptions) {
+	testEndpointAccessabilityWithTLS(pod, "8443", "metrics", allowedConfigs, disallowedConfigs)
+}
+
+func testWebhookEndpoint(pod core.Pod, allowedConfigs, disallowedConfigs []clientTLSOptions) {
+	testEndpointAccessabilityWithTLS(pod, "9443", "", allowedConfigs, disallowedConfigs)
+}
+
+func applyTLSConfig(tlsSecurityProfile *ocpv1.TLSSecurityProfile) {
+	watch, err := StartWatch(sspListerWatcher)
+	Expect(err).ToNot(HaveOccurred())
+	defer watch.Stop()
+
+	updateSsp(func(foundSsp *sspv1beta1.SSP) {
+		foundSsp.Spec.TLSSecurityProfile = tlsSecurityProfile
+	})
+	err = WatchChangesUntil(watch, isStatusDeploying, env.ShortTimeout())
+	Expect(err).ToNot(HaveOccurred(), "SSP status should be deploying.")
+	err = WatchChangesUntil(watch, isStatusDeployed, env.ShortTimeout())
+	Expect(err).ToNot(HaveOccurred(), "SSP status should be deployed.")
+}
+
+func testSettingTLSConfig(pod core.Pod, tlsConfigTestPermutation tlsConfigTestPermutation) {
```

Comment:
You could move/inline this into the definition of your table (L34). That would make it easier to read what the table does.

Reply:
I feel that modern ideas remove this problem and allow for nice separation into small methods.